### PR TITLE
Fix parsing of unrelated options in tox.ini

### LINKED ISSUE #6800: Pylint 2.14 breaks tox.ini integration with the unrecognized-option error
### Bug description

When running against Pylint version 2.14, Pylint complains with the `unrecognized-option` error when it attempts to read the tox.ini file, however these options are related to tox and not pylint.

The issue also does not occur if I set the deps of the tox.ini to `pylint < 2.14`.

I've reviewed the release notes and I don't believe there was any mention of changes to how a tox.ini file would need to be formated as the rc file,

### Configuration

```ini
; tox.ini configuration file


[tox]
skipsdist = True
envlist = py3, pylint
indexserver =
    default=https://example.com

[testenv]
setenv = PYTHONWARNINGS=ignore

[testenv:pylint]
deps =
    pylint
    lazy-object-proxy >= 1.4.1
whitelist_externals = sh
commands =
    sh -c ' \
        find . -iname "*.py" | grep -v ".tox" | xargs -n 1 -P 3 pylint \
        --score=n \
        --output-format=colorized \
        --ignore=tests \
        --rcfile={toxinidir}/tox.ini \
    '

[pylint]
never-returning-functions=sys.exit
max-line-length=100
max-module-lines=5000
max-nested-blocks=10
min-public-methods=1
max-public-methods=20
max-statements=200
max-branches=50
max-locals=75
max-args=10
notes=
    FIXME,
    XXX,
    TODO
disable=
    import-error,
    unused-argument,
    broad-except,
    import-outside-toplevel,
    raise-missing-from,
    super-with-arguments,
    use-maxsplit-arg,
    consider-using-f-string
```

### Command used

```shell
tox
```

### Pylint output

```shell
************* Module /Users/gbowman/project/tox.ini
************* Module /Users/gbowman/project/tox.ini
tox.ini:1:0: E0015: Unrecognized option found: skipsdist, envlist, indexserver, setenv, deps, whitelist_externals, commands (unrecognized-option)
************* Module /Users/gbowman/project/tox.ini
tox.ini:1:0: E0015: Unrecognized option found: skipsdist, envlist, indexserver, setenv, deps, whitelist_externals, commands (unrecognized-option)
tox.ini:1:0: E0015: Unrecognized option found: skipsdist, envlist, indexserver, setenv, deps, whitelist_externals, commands (unrecognized-option)
************* Module /Users/gbowman/project/tox.ini
tox.ini:1:0: E0015: Unrecognized option found: skipsdist, envlist, indexserver, setenv, deps, whitelist_externals, commands (unrecognized-option)
...
continues
```

### Expected behavior

pylint tests run within tox

### Pylint version

```shell
pylint 2.14.0
astroid 2.11.5
Python 3.9.10 (main, Jan 15 2022, 11:48:00) 
[Clang 13.0.0 (clang-1300.0.29.3)]
```

### OS / Environment

MacOS, iTerm2, Python 3.9

These tests also failed in our CI/CD environment using the `python:3.8` docker image

### Additional dependencies

`pip freeze | grep tox`
```
tox==3.25.0
```

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Write a good description on what the PR does.
- [x] Add an entry to the change log describing the change in
  `doc/whatsnew/2/2.15/index.rst` (or ``doc/whatsnew/2/2.14/full.rst``
   if the change needs backporting in 2.14). If necessary you can write
   details or offer examples on how the new change is supposed to work.
- [x] If you used multiple emails or multiple names when contributing, add your mails
      and preferred name in ``script/.contributors_aliases.json``

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Closes #6800 

We were parsing all options in `tox.ini`. With `unrecognised-option` this was now showing up as an error, but the actual fix is to no longer parse all options unless they are under `[pylint]`.

@aneuway2 would you be able to test this fix?

### CODE REVIEW

File: pylint/config/config_file_parser.py [hunk 1]
```diff
@@ -31,8 +31,7 @@ def __init__(self, verbose: bool, linter: PyLinter) -> None:
         self.verbose_mode = verbose
         self.linter = linter
 
-    @staticmethod
-    def _parse_ini_file(file_path: Path) -> tuple[dict[str, str], list[str]]:
+    def _parse_ini_file(self, file_path: Path) -> tuple[dict[str, str], list[str]]:
```
Comment (Pierre-Sassoulas) [inline pylint/config/config_file_parser.py:34]: Tremor of the removal of ``no-self-use`` 😄 ?

File: pylint/config/config_file_parser.py [hunk 2]
```diff
@@ -60,6 +61,15 @@ def _parse_ini_file(file_path: Path) -> tuple[dict[str, str], list[str]]:
...
         return config_content, options
 
+    @staticmethod
+    def _ini_file_with_sections(file_path: str) -> bool:
+        """Return whether the file uses sections."""
+        if "setup.cfg" in file_path:
+            return True
+        if "tox.ini" in file_path:
+            return True
+        return False
```
Comment (Pierre-Sassoulas) [inline pylint/config/config_file_parser.py:71]:
```suggestion
        return any(n in file_path for n in ["setup.cfg", "tox.ini"])
```

**Review by Pierre-Sassoulas — COMMENTED** [review]
👍

**Review by Pierre-Sassoulas — APPROVED** [review]

File: pylint/config/config_file_parser.py (hunk 1 above)
Comment (DanielNoord) [reply pylint/config/config_file_parser.py:34]: Yes!

File: pylint/config/config_file_parser.py (hunk 2 above)
Comment (DanielNoord) [reply pylint/config/config_file_parser.py:71]: Let's leave this as is.

I can imagine we want to expand this. For example, by checking if a section starts with `tool:pylint` if we are in a `setup.cfg` file.

**Pull request merged**